src/saturn_engine/worker/services/tracing/tracer.py

```diff
@@ -97,8 +96,16 @@ def executable_message_attributes(
 def pipeline_message_attributes(
     message: PipelineMessage,
 ) -> Mapping[str, AttributeValue]:
-    return {
-        "saturn.message.id": message.id,
-        "saturn.resources.names": [n for n in message.resource_names if n],
-        "saturn.pipeline.name": message.info.name,
-    } | {f"saturn.message.tags.{k}": v for k, v in message.message.tags.items()}
+    labels_dict: dict[str, AttributeValue] = {}
+    if labels := message.message.metadata.get("labels"):
+        for k, v in labels.items():
+            labels_dict[f"saturn.labels.{k}"] = t.cast(AttributeValue, v)
```

Comment:
I would suggest to just make it a string? `str(v)`

Reply:
We could actually make `labels` dict a `dict[str, str]` (And call `str(v)` from the definition loader), I feel this would make labels easier to deal with and match kubernetes limitation.